### PR TITLE
Happiness chemicals

### PR DESCRIPTION
**Pull request by ArgyleSweatshop**
Adds serotonin, ecstasy, and depressonium. 

Serotonin is intended to cause smiling and laughing, and causes twitching, excessive smiling, and brain damage when ODed. Addiction causes you to cry a lot, and less brain damage.

Ecstasy has a superhigh metabolization rate, to the point where it leaves the body the second it enters it. IT creates 80 units of serotonin and 20 units of depressonium.

Depressonium has a low metabolization rate and causes stamina loss, crying and twitching, and the production of vodka to make the host drunken. This only happens when there is no serotonin to counteract it. It also heals the brain damage caused by serotonin.

Other people are free to fix and balance the terrible mistakes I probably made while coding this.

### CODE REVIEW

**Comment by ArgyleSweatshop** [comment]
I just fixed the first set of errors I found from the "Travis CI build" thing.

**Comment by Robertha89** [comment]
I could read the code, but whats your thoughts regarding addictions and od on this?

**Comment by ArgyleSweatshop** [comment]
The build passed, meaning(hopefully) all errors are gone. PRobably needs some balance testing though.

File: code/modules/reagents/newchem/drugs.dm [hunk 1]
```diff
@@ -389,6 +389,163 @@ datum/reagent/crank/addiction_act_stage4(var/mob/living/M as mob)
...
+
+/datum/reagent/depressonium
+	name = "Depressonium"
+	id = "depressonium"
+	description = "A drug that causes sadness. Pure Sadness."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	metabolization_rate = 0.1
+
+/datum/reagent/seratonin/on_mob_life(var/mob/living/M as mob)
```
Comment (veganzombeh) [inline code/modules/reagents/newchem/drugs.dm]: Shouldn't this be /datum/reagent/depressonium/on_mob_life?

**Comment by Fox-McCloud** [comment]
:-1: 

These aren't really need, they're a partial copy-paste of existing reagents, the recipes make no sense and aren't realistic, despite being realistic reagents, and we already have plenty of illicit drugs (a few of which make you happy) as is.

**Comment by TheDZD** [comment]
Just so we don't add more typos to the code (dealing with them is a nightmare sometimes), it's spelled "serotonin" not "seratonin".

**Comment by ArgyleSweatshop** [comment]
There is no reason to write code different from other reagents as a criterion for including a reagent. I can go through and fix the typos. 

Also, there aren't any chemicals currently that have the specific function of making you "happy".

I also don't really know what criterion you want for making a drug have "realistic recipes".
This game isn't centered around realism. Hell, fliptonium(unrealistic drug), has a recipe consisting of liquid dark matter(unrealistic), chocolate(why?), Ephedrine(Itself a stimulant that isn't a realistic thing), and gin and sonic(Which is a mixture of gin[why?] and meth)
Asking me to make recipes more realistic is silly.

For instance, the logical recipe for serotonin is carbon, hydrogen, nitrogen, and oxygen, but that is obviously way to easy to create.

I'll fix all the typos and such. Anyone know how to fix these issues without making all these commits

**Comment by Fox-McCloud** [comment]
First off, the vast majority of recipes ARE realistic, especially the ones that are strongly based on real-world chemicals; this isn't something that should be broken--a large point of Goonchem is that it's deliberately more realistic so things are fairly logical in nature; it helps improve immersion and generate better RP. Yes, there's exceptions, I don't deny that, but the majority of chemicals are realistic in terms of behaviors and recipes.

Krokodil, Nicotine, and THC all make you happy--yes, they're not designed purely with this in mind, but that's definitely the effect they have on individuals, ICly and mechanically; there's literally no need for overlap in this case.

With the exception of colorful reagent (and maybe 1-2 others), nearly every single chemical in the game has a specific mechanical function for it that has a gameplay element to it; having chemicals for the sake of chemicals (Bay did this with their anti-depressants) just clutters up the code and doesn't serve any real purpose.

Also, copy-pasting effects IS bad as well..generally copy-pasting code is just a bad practice and can lead to messes down the line. It should be reduce whenever possible.

**Comment by ArgyleSweatshop** [comment]
The gameplay element of ecstasy is you can basically poison someone with it and have those negative effects happen fairly quickly. The other two are, essentially, functions of that. 

The difference between this and antidepressants is these actually have an effect. 

I also copypasted zero effects. I hand wrote almost all, if not the entirely of this, although, yes, I pooled from the same effects that all other reagents could.

**Comment by Fox-McCloud** [comment]
A reagent that adds literally 100 total reagents to your system ever life proc. This is neither balanced, nor logical.

Sorry, this isn't ever going to get a :+1: from me, as many of my points still stand.

**Comment by ArgyleSweatshop** [comment]
The serotonin metabolizes at a rate of 4.5 units a cycle. It won't last long in your body, which is why ecstasy produces so much of it. Since you've admitted you'll never approve it, I don't see a necessity to respond to any of your concerns.

**Comment by tigercat2000** [comment]
I actually like the serotonin and depressonium, aaalthough the depress one could be better named.

As for the ecstasy.. meh..

**Comment by ArgyleSweatshop** [comment]
It is essentially how ecstasy works in real life. IT doesn't do much by itself, but it causes your brain to mass dump oxytocin(I didn't want to use oxytocin, because oxytocin is mainly only produced by the body when you... ahem... have sex, and I didn't want to imagine getting an erection every time someone hugged you). You then spiral into depression. I could remove ecstasy and give a recipe for depressonium, if its desired.

**Comment by Krausus** [comment]
I'm pretty sure we axed a number of mood-affecting drugs during the transition to Goonchem because, as it turns out, we aren't high RP so nobody actually cares about them.

These, though, seem to be less about affecting mood and more about having a nice big "I win" button in chemical form. Ecstacy is trivial to make if you have access to chemistry equipment, a drop causes instant serotonin overdose, and serotonin overdose has a 1/3 chance per tick of flooring someone for **fifteen seconds.** Oh, and if the virtually guaranteed 20+ seconds of weakness (per drop) aren't enough to finish someone off, then by the time they can stand again, they'll have the depressonium applying weakness with a roughly 50% uptime.

Ecstacy is absolutely ridiculous.

You could remove ecstacy... but then serotonin will basically be impossible to obtain, since it requires bath salts and we currently don't have a source of mugwort available to anyone but wizards. That's not actually supposed to be the case, but nobody's gotten around to fixing it.

Quite frankly, these don't seem well thought out in general.

Also, serotonin has two `overdose_process` procs, and ecstasy has two `chemical_reaction`s.

> Other people are free to fix and balance the terrible mistakes I probably made while coding this.

Working out the kinks *before* you make a pull request is kinda important if you want to be taken seriously. At the very least, you should make sure your code actually compiles first. As it stands, though, there's already a lot of code in the game that needs fixing, and none of the people working on it want to see more added without good reason.

**Comment by ArgyleSweatshop** [comment]
If I remove ecstasy, add a recipe for depressonium, and balance out serotonin(and fix the recipe), Might that be enough to be approved? 

I also note that people seem to like depressonium the most(it seems more unique, IMO), so if only adding that would be good enough, would that be approved?

Because if none of these chemicals will be approved under any circumstance, there isn't much of a point

**Comment by Fox-McCloud** [comment]
Still doesn't have my support, sorry.

**Pull request closed without merging**